MW2-208Project/MW2-208Project/Utils.cpp

```diff
@@ -105,8 +105,50 @@ bool Utils::ValidAddy(DWORD_PTR Addy)
 	return (Addy > modBase && Addy < modSiz) || (Addy > curDLL && Addy < curSiz);
 }
 
+int Utils::UnprotectPE()
+{
+
+	// unprotect the entire PE
+	HMODULE hModule = GetModuleHandle( nullptr );
+
+	if ( !hModule ) {
+
+		return ERROR_MOD_NOT_FOUND;
+	}
+
+	const auto header = reinterpret_cast< PIMAGE_DOS_HEADER >( hModule );
+
+	const auto ntHeader = reinterpret_cast< PIMAGE_NT_HEADERS >( reinterpret_cast< uintptr_t >( hModule ) + header->e_lfanew );
+
+	const SIZE_T size = ntHeader->OptionalHeader.SizeOfImage;
+
+	DWORD oProtect;
+
+	VirtualProtect( static_cast< LPVOID >( hModule ), size, PAGE_EXECUTE_READWRITE, &oProtect );
+
+	return ERROR_SUCCESS;
+}
+
 void Utils::Init(HMODULE thisModule)
 {
+	//credits to https://github.com/CamxxCore/PatchIWNet
+	if (UnprotectPE() != ERROR_SUCCESS)
+		return;
```

Comment:
Completely unnecessary. Even though VAC is quite dumb in MW2, we shouldn't permanently unprotect the entire executable.

Use `PatchAddy<Type>(Address, Value);` to automatically unprotect a single address space and patch the executable.

```suggestion
void Utils::Init(HMODULE thisModule)
{
```